Source/CarthageKit/Xcode.swift

```diff
@@ -234,7 +234,9 @@ public func schemesInProject(project: ProjectLocator) -> ColdSignal<String> {
 		// xcodebuild has a bug where xcodebuild -list can sometimes hang
 		// indefinitely on projects that don't share any schemes, so
 		// automatically bail out if it looks like that's happening.
-		.timeoutWithError(CarthageError.NoSharedSchemes(project).error, afterInterval: 4, onScheduler: QueueScheduler())
+		// xcodebuild may also take a while to execute;
+		// see https://github.com/carthage/carthage/issues/299
```

Comment:
I think it's understood that we could create false negatives here. I don't think it needs a comment.